--- src/money.3.c	2017-03-28 01:11:35.000000000 +0000
+++ src/money.4.c	2017-03-28 01:11:35.000000000 +0000
@@ -1,22 +1,27 @@
 #include <stdlib.h>
 #include "money.h"
 
+struct Money
+{
+    int amount;
+};
+
 Money *money_create(int amount, char *currency)
 {
     return NULL;
 }
 
 int money_amount(Money * m)
 {
-    return 0;
+    return m->amount;
 }
 
 char *money_currency(Money * m)
 {
     return NULL;
 }
 
 void money_free(Money * m)
 {
     return;
 }
